Tiered List HOC/Pattern (#5212)
canonical/vanilla-framework: templates/docs/patterns/split-list/index.md

<div class="embedded-example"><a href="/docs/examples/patterns/split-list/50-50-desktop-with-description/" class="js-example">
View example of the matrix pattern
</a></div>

## 50/50 on tablet w/o description

<div class="embedded-example"><a href="/docs/examples/patterns/split-list/50-50-tablet-no-description/" class="js-example">
View example of the matrix pattern
</a></div>

## 50/50 on tablet w/ description

<div class="embedded-example"><a href="/docs/examples/patterns/split-list/50-50-tablet-with-description/" class="js-example">
View example of the matrix pattern
</a></div>

## 50/50 w/ description

<div class="embedded-example"><a href="/docs/examples/patterns/split-list/50-50-with-description/" class="js-example">
View example of the matrix pattern
</a></div>

## Full-width w/o description

<div class="embedded-example"><a href="/docs/examples/patterns/split-list/full-width-no-description/" class="js-example">
View example of the matrix pattern
</a></div>

## Full-width w/ description

<div class="embedded-example"><a href="/docs/examples/patterns/split-list/full-width-with-description/" class="js-example">
View example of the matrix pattern
</a></div>

## Import

> To import and use this Macro in your project, follow the steps on the

Reviewer: As far as I'm aware, just importing and using the macro will not also include the SCSS used by the macro - and this macro requires `vf-p-headings`.  

We should probably include instructions for doing that here, similar to the [other components](https://vanilla-framework-5212.demos.haus/docs/patterns/slider#import)

> [Macro installation guide](TODO).

Reviewer 2: If there is no installation guide yet, let's not document it as such.

At this point it technically doesn't matter if it's a macro or not (if devs can't use it). We also don't really have a practice of documenting these macros, so need to figure that out.

I'd just suggest documenting it as a standard HTML for now, with examples to copy (internally using macros of course). And we can build on top of that when we have means to install and document those macros.

Reviewer 2: Also, this link is 404, causing linkchecker to fail.

---
wrapper_template: '_layouts/docs.html'
context:
  title: Split list | Patterns
---

The split list pattern is used to easily implement a list of paired titles and
descriptions, complete with a top-level title and description. An optional CTA
block is placed at the bottom of the content as well. Depending on the size and
composition of your content, you can choose from a variety of split list
layouts.

## 50/50 on desktop w/ description

This variant contains a top-level description which is presented side-by-side
with its title on desktop screen sizes.

<div class="embedded-example"><a href="/docs/examples/patterns/split-list/50-50-desktop-with-description/" class="js-example">

> View example of the matrix pattern

Reviewer: ```suggestion
View example of the split-list pattern
```

</a></div>

## 50/50 on tablet w/o description

This variant does not contain a top-level description and its child list is
presented with its titles side-by-side with its descriptions on tablet screen
sizes.

> <div class="embedded-example"><a href="/docs/examples/patterns/split-list/50-50-tablet-no-description/" class="js-example">

Reviewer: Not a show-stopper, and not sure if this was a stylistic choice or not, but there are some places throughout these examples that could use a bit more whitespace control to get rid of extra newlines
![image](https://github.com/user-attachments/assets/bb179786-c68a-46a8-a559-22096de5aa0d)

View example of the matrix pattern
</a></div>

## 50/50 on tablet w/ description

This variant contains a top-level description and its child list is presented
with its titles side-by-side with its descriptions on tablet screen sizes.

<div class="embedded-example"><a href="/docs/examples/patterns/split-list/50-50-tablet-with-description/" class="js-example">

> View example of the matrix pattern

Reviewer: Backup / fallback text for these examples needs to be updated to reflect split list pattern, not matrix pattern
```suggestion
View example of the split-list pattern
```

</a></div>

## 50/50 w/ description

This variant contains a top-level description. Its title and description are
presented side-by-side on desktop screen sizes, and its child list is presented
side-by-side on tablet screen sizes.

<div class="embedded-example"><a href="/docs/examples/patterns/split-list/50-50-with-description/" class="js-example">

> View example of the matrix pattern

Reviewer: ```suggestion
View example of the split-list pattern
```

</a></div>

> ## Full-width w/o description

Reviewer: These "without <slot>" and "with <slot>" examples make me wonder if Hero should have something similar as well. 

It's rather verbose in the current structure of our docs (having a whole new example that is the same as the previous one but with/without one DOM node), but it's the best way of doing this right now.

This variant does not contain a top-level description, and both its title and
child list are presented full-width on desktop and tablet screen sizes
respectively.

<div class="embedded-example"><a href="/docs/examples/patterns/split-list/full-width-no-description/" class="js-example">

> View example of the matrix pattern

Reviewer: ```suggestion
View example of the split-list pattern
```

</a></div>

## Full-width w/ description

> This variant contains a top-level description, and its title, description, and
> child list are presented full-width on desktop and tablet screen sizes
> respectively.

Reviewer: This feels a bit wordy. Maybe something like:

```suggestion
This variant contains a top-level description. Its contents are presented full-width on desktop and tablet screen sizes.
```

<div class="embedded-example"><a href="/docs/examples/patterns/split-list/full-width-with-description/" class="js-example">

> View example of the matrix pattern

Reviewer: ```suggestion
View example of the split-list pattern
```
